src/applications/personalization/appointments/actions/index.js

```diff
@@ -1,27 +1,216 @@
-          data: res.data.attributes.appointments,
-        }),
-      )
-      .catch(err => {
-        dispatch({
-          type: FETCH_APPOINTMENTS_FAILURE,
-          err,
-        });
+    let facilitiesLookup = {};
+    let facilitiesResponse;
+    let vaAppointments = [];
+    let ccAppointments = [];
+
+    const startOfToday = moment()
+      .startOf('day')
+      .toISOString();
+
+    try {
+      if (environment.isLocalhost()) {
+        vaAppointments = MOCK_VA_APPOINTENTS;
+        ccAppointments = MOCK_CC_APPOINTMENTS;
+      } else {
+        vaAppointments = await apiRequest(
+          `/vaos/v0/appointments?start_date=${startOfToday}&type=va`,
+        );
+        ccAppointments = await apiRequest(
+          `/vaos/v0/appointments?start_date=${startOfToday}&type=cc`,
+        );
+      }
+      const facilityIDs = uniq(
+        map(
+          vaAppointments,
+          appointment =>
+            getStagingID(appointment?.attributes?.sta6aid)
+              ? `vha_${getStagingID(appointment?.attributes?.sta6aid)}`
+              : undefined,
+        ),
+      );
+
+      if (environment.isLocalhost()) {
+        facilitiesResponse = MOCK_FACILITIES;
+      } else {
+        facilitiesResponse = await apiRequest(
+          `${environment.API_URL}/v1/facilities/va?ids=${facilityIDs.join(
+            ',',
+          )}`,
+        );
+      }
+
+      facilitiesLookup = reduce(
+        facilitiesResponse?.data,
+        (lookup, facility) => {
+          const facilityID = replace(facility?.id, 'vha_', '');
+          // eslint-disable-next-line no-param-reassign
+          lookup[facilityID] = facility;
+          return lookup;
+        },
+        {},
+      );
+    } catch (error) {
+      dispatch({
+        type: FETCH_CONFIRMED_FUTURE_APPOINTMENTS_FAILED,
       });
+    }
+
+    const formattedVAAppointments = reduce(
+      vaAppointments,
+      (accumulator, appointment) => {
+        const startDate = moment(appointment?.attributes?.startDate);
+        const now = moment();
+        const appointmentStatus =
+          appointment?.attributes?.vdsAppointments?.[0]?.currentStatus;
+
+        // Past appointments should be filtered out already on the api side, this is a safe guard.
+        if (startDate.isBefore(now)) {
+          return accumulator;
+        }
+
+        // Hide cancelled or no-show appointments
+        if (FUTURE_APPOINTMENTS_HIDDEN_SET.has(appointmentStatus)) {
+          return accumulator;
+        }
+
+        // Derive the facility.
+        const facility =
+          facilitiesLookup[getStagingID(appointment?.attributes?.facilityId)];
+
+        accumulator.push({
+          additionalInfo: getAdditionalInfo(appointment),
+          facility,
+          id: appointment?.id,
+          isVideo: !isEmpty(appointment?.attributes?.vvsAppointments),
+          providerName: facility?.attributes?.name,
+          startsAt: startDate.toISOString(),
```

Comment:
@jbalboni I have seen some vaos logic where we are setting the startTime of video appointments to `vvsAppointments.dateTime`. Do you know if I need to update this logic to reflect that, or can we safely assume that the `startDate` is a safe value?

Reply:
You should be able to safely use `startDate`. We probably need to clean that up in VAOS.

Reply:
Thanks for confirming so quickly! 💯